### PR TITLE
XRT-577 Update to support string and array data hashing

### PR DESCRIPTION
**Pull request by SteveOss**

### CODE REVIEW

File: include/iot/hash.h [hunk 1]
```diff
@@ -21,11 +21,20 @@ extern "C" {
...
- * @param str Pointer to string - Key to hash
+ * @param str Pointer to string to be hashed
  * @return    Hash value
  */
 extern uint32_t iot_hash (const char * str);
 
+/**
+ * @brief Bernstein djb2 hash function (Version 2)
+ *
+ * @param date Pointer to data to be hashed
```
Comment (BinduRao2018) [inline include/iot/hash.h:32]: typo - should be data

Comment (SteveOss) [reply include/iot/hash.h]: Have pushed fix

**Review by BinduRao2018 — APPROVED** [review]
LGTM

**Comment by BinduRao2018** [comment]
@SteveOss, Could you please merge, as I don't have the permission?

**Pull request merged**